frontends/p4/fromv1.0/converters.cpp

```diff
@@ -716,6 +700,45 @@ class ComputeCallGraph : public Inspector {
     }
 };
 
+/// Table call graph should be built after the control call graph is built.
+/// In the case that the program contains an unused control block, the
+/// table invocation in the unused control block should not be considered.
+class ComputeTableCallGraph : public Inspector {
+    ProgramStructure *structure;
+
+ public:
+    explicit ComputeTableCallGraph(ProgramStructure *structure) : structure(structure) {
+        CHECK_NULL(structure);
+        setName("ComputeCallGraph");
+    }
+
+    void postorder(const IR::Apply *apply) override {
+        LOG3("Scanning " << apply->name);
+        auto tbl = structure->tables.get(apply->name.name);
+        if (tbl == nullptr)
+            ::error("Could not find table %1%", apply->name);
+        auto parent = findContext<IR::V1Control>();
+        BUG_CHECK(parent != nullptr, "%1%: Apply not within a control block?", apply);
```

Comment:
Is this a BUG_CHECK because it would have been an error in the parser? Otherwise, seems like a program error.